pkg/resource/deploy/plan.go

```diff
@@ -60,6 +60,17 @@ type PlanSummary interface {
 	Sames() map[resource.URN]bool
 }
 
+// PlanPendingOperationsError is an error returned from `NewPlan` if there exist pending operations in the
+// snapshot that we are preparing to operate upon. The engine does not allow any operations to be pending
+// when operating on a snapshot.
+type PlanPendingOperationsError struct {
+	Operations []resource.Operation
+}
+
+func (p PlanPendingOperationsError) Error() string {
+	return "one or more operations are currently pending"
```

Comment:
Nit: can we standardize on `InFlight` or `Pending`? My  preference is for the latter, but all I really care about is that we pick one term in strings that might end up being displayed to a user.

Reply:
Yeah, I think I'll go with `Pending` - I like that one more too.